packages/@react-stately/layout/src/ListLayout.ts

```diff
@@ -136,22 +142,24 @@ export class ListLayout<T> extends Layout<Node<T>> implements KeyboardDelegate {
       nodes.push(layoutNode);
     }
 
-    if (nodes.length === 0) {
-      let rect = new Rect(0, y, this.virtualizer.visibleRect.width, this.rowHeight || this.estimatedRowHeight);
-      let placeholder = new LayoutInfo('placeholder', 'placeholder', rect);
-      this.layoutInfos.set('placeholder', placeholder);
-      nodes.push({layoutInfo: placeholder});
-      y = placeholder.rect.maxY;
-    }
-
     if (this.isLoading) {
-      let rect = new Rect(0, y, this.virtualizer.visibleRect.width, 40);
+      let rect = new Rect(0, y, this.virtualizer.visibleRect.width,
+        this.loaderHeight ?? this.virtualizer.visibleRect.height);
       let loader = new LayoutInfo('loader', 'loader', rect);
       this.layoutInfos.set('loader', loader);
       nodes.push({layoutInfo: loader});
       y = loader.rect.maxY;
     }
 
+    if (nodes.length === 0) {
+      let rect = new Rect(0, y, this.virtualizer.visibleRect.width,
+        this.placeholderHeight ?? this.virtualizer.visibleRect.height);
```

Comment:
had to add this because the placeholder had the same unbounded resizing, but unsure if this the solution we'd want (is there some other value we can key off? was thinking about using `estimatedRowHeight`, however, list view uses that too but needs to have the placeholder span `visibleRect.height`)